Common/Extensions.cs

```diff
@@ -40,6 +42,36 @@ namespace QuantConnect
     /// </summary>
     public static class Extensions
     {
+        /// <summary>
+        /// Given a type will create a new instance using the parameterless constructor
+        /// and assert the type implements <see cref="IBaseData"/>
+        /// </summary>
+        /// <remarks>One of the objectives of this method is to normalize the creation of the
+        /// IBaseData instances while reducing code duplication</remarks>
+        public static IBaseData GetBaseDataInstance(this Type type)
+        {
+            var objectActivator = ObjectActivator.GetActivator(type);
+            if (objectActivator == null)
+            {
+                throw new ArgumentException($"Data type \'{type.Name}\' missing parameterless constructor " +
+                    $"E.g. public {type.Name}() {{ }}");
+            }
+
+            var instance = objectActivator.Invoke(new object[] { type });
```

Comment:
If `objectActivator` is pointing towards a parameterless constructor, then why are we passing in the `type`?

Reply:
Good question 😄, truth is I just copied existing implementation, didn't change anything here. Don't think its worth changing in _this_ PR though.

- Tested the following and it fails for python custom data, so maybe it's being used by PythonNet somehow:
```
        public static Func<object> GetParameterlessActivator(Type dataType)
        {
            lock (_lock)
            {
                // if we already have it, just use it
                Func<object> factory;
                if (_parameterlessActivatorsByType.TryGetValue(dataType, out factory))
                {
                    return factory;
                }
                
                var ctor = dataType.GetConstructor(new Type[] { });

                //User has forgotten to include a parameterless constructor:
                if (ctor == null) return null;

                var newExp = Expression.New(ctor);
                var lambda = Expression.Lambda(typeof(Func<object>), newExp);
                factory = (Func<object>)lambda.Compile();

                // save it for later
                _parameterlessActivatorsByType.Add(dataType, factory);

                return factory;
            }
        }
```
PS: the original implementation is from 5 years ago Mike 😄